src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -597,7 +597,18 @@ private static FirebaseOptions getOptionsFromEnvironment() throws IOException {
       parser = jsonFactory.createJsonParser(reader);
     }
     parser.parseAndClose(builder);
-    builder.setCredentials(GoogleCredentials.getApplicationDefault());
+    builder.setCredentials(getApplicationDefaultOrEmptyCredentials());
     return builder.build();
   }
+
```

Comment:
This is done to enforce that we don't error out, but proceed with empty credentials if we were unable to fetch default credentials.

Reply:
Most of the time it is an error to deploy the SDK where ADC is not available. I think emulator is probably going to be the only exception. Can we also check if the emulator is set up (perhaps by checking the env variable), and if not retain the existing behavior?

Reply:
The behavior in this PR is consistent with the Python Admin SDK and with the approved internal design doc. To reiterate, the reason behind this is 1) Developers may use RTDB config (databaseUrl) instead of environment variables. It is impossible to detect that here, and 2) poking into product-specific config / env vars in FirebaseApp introduces circular dependencies and in general make the code more messy.

However, we should make sure it is still an error to run the SDK when ADC is not available. It does not have to throw an exception here -- a delayed exception when we actually try to connect to RTDB/Firestore/Storage/etc. is also acceptable per the design doc. @sarajmunjal , is this already the case? If so, please add comments / tests to clarify. If not, please make it so.

Reply:
The behavior as implemented here is not consistent with the Python implementation. We don't have the deferred credential look up feature here. May be implement a new `DeferredApplicationDefaultCredentials` implementation which provides that deferred behavior.

Reply:
Done!

Reply:
I think with the updated code the behavior is now inline with the Python admin SDK. (And sorry I misunderstood you earlier -- your comment makes a lot of sense.)